xbmc/music/MusicDatabase.cpp

```diff
@@ -10501,6 +10784,11 @@ void CMusicDatabase::SetPropertiesFromAlbum(CFileItem& item, const CAlbum& album
     item.SetProperty("album_userrating", album.iUserrating);
   if (album.iVotes > 0)
     item.SetProperty("album_votes", album.iVotes);
+
+  if (album.bBoxedSet)
+    item.SetProperty("album_bBoxedSet", true); //! @todo Set listItem the same way as totalDiscs ??
+  item.SetProperty("album_totaldiscs", album.iTotalDiscs);
+  item.GetMusicInfoTag()->SetTotalDiscs(album.iTotalDiscs); // sets for ListItem.TotalDiscs
```

Comment:
Is this being sneeky and giving each song in an item list a disc total value? 
I don't think we set tag data here unless you have a reason. This is setting album details in the item property map for an item that is a song (hence the song data is in the musicinfotag). It isn't doing any harm because the song doesn't have a different value held in that part of the tag, but it is adding confusion where people already get confused about when to use a property map and when to use the tag structure.

Reply:
From the modified confluence skin I'm using, no it doesn't appear to set it for a song.
[<a href="https://ibb.co/vv1hQGw"><img src="https://i.ibb.co/vv1hQGw/Screenshot-from-2019-10-23-10-17-27.png" alt="Screenshot-from-2019-10-23-10-17-27" border="0"></a>](url) but that is, IMHO, useful information to display there.  That means adding iTotalDiscs to the songview so that `GetFileItemFromDataset` can set it there as I feel it's useful info that should be displayed there.  Much better if in that screenshot I know I am on disc 2 of a possible 3 - wish I was better with the skinning engine !

Reply:
> That means adding iTotalDiscs to the songview

No hang on, don't do that there is no need - all the album info is available for each song item you just have to use the correct info label

Reply:
Then there is also the question of what happens should I get to my todo list entry that includes adding  processing of the TPOS/DISCTOTAL tag - the number of discs on that release (user may have less in the their library if they have partial albums, and a surprizing number do)

Reply:
Grrr Dave !!  Now you're just complicating things again !! :) 

Some of my stuff technically has less discs than it should have because a few releases also included a DVD which I have added to the video library under a custom node.  So I have albums with 4 or 5 discs where there _should_ be another one really but it's not in the music library.  Guess that would open up the possibility to add a filter `select * from albumview where iDiscTotal != iReleaseDiscs` to show albums that are missing a disc or 3 !!
>all the album info is available for each song item you just have to use the correct info label

Is it ?  It all appears to be set from the songview.  Maybe I've not looked hard enough but I can't see this stuff getting set for songs anywhere else ???

Reply:
Me complicate.... never :)
I want to see what users do with this new feature before (and if)  I add yet more disc count related data.  Unless you suddenly have a deep desire to include that tag in your work here too? Up to you, I'm not saying do it but I wouldn't stop you either. 
BTW I think what you have done is far more useful/interesting than just one more bit of  tag data saying this song from an album only partially in your library actually came from a release with 5 discs etc. 

The property map of the current item gets loaded with album data, just like it gets loaded with all the art work for related album and artist. 

The songview is already bloated and the songs fileitem list notoriously slow to fill, including a repeat of the album data in each song item would make that worse. What is in the songview is just what may be shown on the list view (values for many  songs on one screen), it includes album title because it is something we may want to sort by, but not the other album fields. I really don't think there is a need to show disc total in the list (for many songs at a time), showing it for the song the cursor is on is sufficient (along with other album data).  Hence 
don't add the album data iDiscTotal to the songview.

Reply:
I think it needs to be there.  Otherwise I can't get a view like this
<a href="https://ibb.co/7vKK52k"><img src="https://i.ibb.co/7vKK52k/Screenshot-from-2019-10-23-14-05-24.png" alt="Screenshot-from-2019-10-23-14-05-24" border="0"></a>
If I don't set it in there, I just get 'Disc 1 of 0' which is clearly wrong.  The property map of the item doesn't appear to contain the right information in that view but that could be my non-existent skinning skills at fault although I think `Listitem.Property(Album_Totaldiscs)` should work, but it doesn't.  `ListItem.TotalDiscs` works in every view _except_ the songview where it returns nothing.

So, using `listitem.property(Album_Totaldiscs)` or `listitem.TotalDiscs` works in DialogMusicInfo.xml and any view that contains albums.  Neither of those work in the songview unless I specifically set them in GetSongFromDataset.  But I need iTotalDiscs in the view to do that otherwise there is nowhere to get the info from.

Suggestions gratefully received ! :)

  BTW, the horribly bad three disc overlay on the cover is courtesy of a `<visible>ListItem.IsBoxset</visible>` tag.  There is no 'of' in strings.po though so that is hard-coded although I could have used ' / ' or something similar.  I like how it looks though.

Reply:
Ah, there is a bug..... :(
The route taken in and about songview nodes effects is the album (and artist) properties get populated or not.  Obviously skins have not been using it, or have worked around it if they saw issues as I don't think anyone mentioned this before.

I would rather see that fixed, and all album props available, than mess with songview to get you that x of y immediately. But it is a non-trival bug from my inital run through it, will my todo list ever end?

Reply:
>Ah, there is a bug..... :(

No....Surely not  !! :)

>will my todo list ever end?

Unlikely, especially as you keep fixing things in other areas as well - still, it keeps the mind stimulated !  Guess I had better revert my local copy then and live without x of y for now as I have also found a (fortunately trivial) issue with my code.  _Unfortunately_, it affects my 'daily driver' which needs python2 for BBC iPlayer to work :(  Found it when I went back to my radio helper addon and turned on it's local library support.  It fell over with a MySQL error when updating a song comment.  There is/was a ' missing from `strDiscSubtitle = '%s'` in `CMusicDatabase::UpdateSong`.

Reply:
I worked around the bug by adding a hidden button to the view and then used its <onfocus> property to set a window property that contains the total number of discs and then read that back on the disc and song views.  It's not as slick as having the actual ListItem available there, but it works so I can still have disc X of Y without having to touch songview :)